packages/core/src/components/tree/treeNode.tsx

```diff
@@ -78,6 +84,14 @@ export interface ITreeNodeProps<T = {}> extends ITreeNode<T> {
     path: number[];
 }
 
+interface ITreeNodeMouseEventHandlers {
```

Comment:
@adidahiya I had to introduce this unexported interface in order to avoid typescript error at `eventHandlers = {}`
```
 Type '{}' is not assignable to type '{ onClick: (e: MouseEvent<HTMLDivElement>) => void; ... }
```
If there is a way to avoid this error, let me know

Reply:
no, you shouldn't need this interface. try

```ts
const eventHandlers = disabled === true ? {} : {
    onClick: this.handleClick,
    onContextMenu: this.handleContextMenu,
    onDoubleClick: this.handleDoubleClick,
    onMouseEnter: this.handleMouseEnter,
    onMouseLeave: this.handleMouseLeave,
};
```

and then the compiler ought to infer the types correctly without an explicit typedef

Reply:
Fixed, thanks. Btw. linter decided to do this if statement multi-line :grinning: